packages/BetterPhpDocParser/PhpDocManipulator/PhpDocTypeChanger.php

```diff
@@ -65,6 +65,9 @@ public function changeVarType(PhpDocInfo $phpDocInfo, Type $newType): void
             // only change type
             $currentVarTagValueNode->type = $newPHPStanPhpDocType;
         } else {
+            // remove @inheritDoc to not end up with both @inheritDoc and @var - it doesn't make sense
+            $phpDocInfo->removeInheritDoc();
```

Comment:
What if the param `@var` type is more precise?

What else is inherited by the `@inheritdoc`? This change might be removing very important information that is used by PHPStan. 

IMO it would be better to avoid it here. If you think it might be useful, there could a separated rule to remove dead inherited doc. We have similar rules for removing useless param/var/return tags that are already in PHP type declaration.

Reply:
Yeah, it might be actually a good idea to revert this change.